hs/src/Reach/Simulator/Core.hs

```diff
@@ -514,26 +528,21 @@ instance Interp DLExpr where
       consensusPayout acc dlPayAmnt
       return v
     DLE_TokenNew _at dltokennew -> do
-      ledgerNewToken simContract dltokennew
-      return V_Null
+      (g, _) <- getState
+      let accIdMax = (e_naccid g) - 1
+      tokId <- ledgerNewTokens accIdMax dltokennew
```

Comment:
@jeapostrophe this code creates ONE new token. if there are currently 3 accounts in existence, it will add the ONE new token just created to the `Wallet` of all 3 accounts in existence.

Reply:
Ah, for my stupidity, I think you should do something like:
```
tokId <- allocANewTokenId
addTheTokenToTheContractWallet tokId dltokennew
return $ V_Token tokId
```
and then when you look up balances, you use `fromMaybe 0 $ M.lookup tokId balances` and you don't need to explicitly add it ever